frontend/src/views/Game.tsx

```diff
@@ -102,18 +139,32 @@ function GamePage() {
     window.dispatchEvent(event);
   };
 
+  // Send notification if submission result is correct and currentUser is set.
+  const checkSendTestCorrectNotification = (submissionParam: SubmissionResult) => {
+    if (submissionParam.status === 'SUCCESS' && currentUser) {
+      const notificationBody: string = JSON.stringify({
```

Comment:
No changes needed but in the future I was thinking this notification could possibly be triggered on the backend (user submits solution, backend scores the submission and sends any relevant socket messages through NotificationService). However I can imagine maybe other notifications like code typing speed (which could be hard to calculate/sync on the backend) being done from the frontend

Reply:
That's fair, I will note that if we do the "Run Test" code in the frontend through JavaScript that @helenli8181 is working on, then this could potentially be done completely in the frontend -- but I think backend also works, and we can figure that out in the future.